### PR TITLE
Add a privacy consideration note for issue #330.

### LINKED ISSUE #330: CanMakePaymentEvent handling algorithm unclear
The current specification for `CanMakePaymentEvent` handling in the Payment Handler API is unclear in light of the `canMakePayment`/`hasEnrolledInstrument` change in Payment Request API [1][2].

First, the new `canMakePayment` is meant to represent the existence of support without checking available instrument. So perhaps `CanMakePaymentEvent` should not be dispatched at all in this case.

Second, if `CanMakePaymentEvent` should be dispatched for `hasEnrolledInstrument`, the current algorithm [3] is unclear about the return value when the steps are terminated early:
> 4.3 Handling a CanMakePaymentEvent
Upon receiving a PaymentRequest, the user agent MUST run the following steps:
> 1. If user agent settings prohibit usage of CanMakePaymentEvent (e.g., in private browsing mode), terminate these steps.
> 2. Let registration be a ServiceWorkerRegistration.
> 3. If registration is not found, terminate these steps.

Chrome's current implementation returns `true` if the early termination is triggered from step 1 or if it is triggered for just-in-time installable payment handlers from step 3. This will be the wrong answer if the website's intention is to check `hasEnrolledInstrument`.

I think it will make more sense to clarify in the spec these two changes:

- `CanMakePaymentEvent` is only dispatched when website queries `hasEnrolledInstrument`
- Implementations should return `false` when `CanMakePaymentEvent` cannot be dispatched, either due to private browsing mode or if the payment handler is not installed.

[1] https://github.com/w3c/payment-request/pull/806
[2] https://lists.w3.org/Archives/Public/public-payments-wg/2018Nov/0018.html
[3] https://w3c.github.io/payment-handler/#handling-a-canmakepaymentevent

### PR DESCRIPTION
**Pull request by danyao**
closes #330 

The following tasks have been completed:

 * [ ] web platform tests (link)
 * [ ] MDN Docs added (link)

Implementation commitment:

 * [ ] Safari (link to issue)
 * [ ] Chrome (link to issue)
 * [ ] Firefox (link to issue)
 * [ ] Edge (public signal)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-handler/pull/331.html" title="Last updated on Jan 30, 2019, 3:51 PM UTC (203236b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-handler/331/be353e8...203236b.html" title="Last updated on Jan 30, 2019, 3:51 PM UTC (203236b)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2235,7 +2235,10 @@ <h2>
           <li>The <a>CanMakePaymentEvent</a> event should not be fired in
           private browsing mode. The user agent should behave as if
             <a data-lt="CanMakePaymentEvent.respondWith()">respondWith()</a>
-            was called with <code>true</code>.
+            was called with <code>false</code>. We acknowledge a consequent
+            risk: if a merchant controls both the origin of the Payment Request
```
Comment (ianbjacobs) [inline index.html]:
```suggestion
            risk: if an entity controls both the origin of the Payment Request
```

File: index.html [hunk 2]
```diff
@@ -2235,7 +2235,10 @@ <h2>
...
+            API call and the payment handler, it is possible to deduce from the
```
Comment (ianbjacobs) [inline index.html]:
```suggestion
            API call and the origin of the payment handler, that entity may be able to deduce
```

File: index.html [hunk 3]
```diff
@@ -2235,7 +2235,10 @@ <h2>
...
+            result that the user may be in private browsing mode.
```
Comment (ianbjacobs) [inline index.html]:
```suggestion
            that the user may be in private browsing mode.
```

**Review by ianbjacobs — COMMENTED** [review]
Thanks, Danyao! I have made some minor editorial suggestions. I used the new GitHub tool (shown to me by Marcos) so if you are happy with the changes you can easily incorporate them.

**Comment by danyao** [comment]
Thanks for the review! I incorporated your changes (the web interface is easy to use!). PTAL.

**Review by ianbjacobs — APPROVED** [review]

**Pull request merged**